packages/middleware/src/handlers/prepareApiFunction/index.ts

```diff
@@ -29,6 +29,40 @@ export function prepareApiFunction(
       res,
       extensions,
       customQueries,
+      integrations,
+      getIntegration: (integrationKey: string) => {
+        if (!integrations[integrationKey]) {
+          const keys = Object.keys(integrations);
+          throw new Error(
+            `The specified integration key "${integrationKey}" was not found. Available integration keys are: ${keys}. Please ensure you're using the correct key or add the necessary integration configuration.`
+          );
+        }
+
+        const {
+          apiClient: innerApiClient,
+          configuration: innerConfiguration,
+          extensions: innerExtensions,
+          customQueries: innerCustomQueries = {},
+          initConfig: innerInitConfig,
+        } = integrations[integrationKey];
+
+        const innerMiddlewareContext: MiddlewareContext = {
+          ...middlewareContext,
+          extensions: innerExtensions,
+          customQueries: innerCustomQueries,
+        };
+
+        const createInnerApiClient = innerApiClient.createApiClient.bind({
+          middleware: innerMiddlewareContext,
+        });
+
+        const apiClientInstance = createInnerApiClient({
+          ...innerConfiguration,
+          ...innerInitConfig,
```

Comment:
Do we preserve the methods exposed on `context.api` or do we not have to worry about them here?

Reply:
Context.api method will be generated during innerCreateApiClient (same behaviour as with previous createApiClient), so nothing changes here